index.js

```diff
@@ -26,19 +29,30 @@ module.exports = function (src, dest, opts) {
 
 	opts = objectAssign({overwrite: true}, opts);
 
-	return new Promise(function startRead(resolve, reject) {
-		var read = fs.createReadStream(src);
-
-		read.on('error', function (err) {
-			reject(new CpFileError('cannot read from `' + src + '`: ' + err.message, err));
-		});
+	var progress = {
+		stat: null,
+		emitter: new EventEmitter()
+	};
```

Comment:
Yes, but IMO it is easier to read and to maintain, e.g. we may emit other events in the future...